packages/core/src/components/segmented-control/segmentedControl.tsx

```diff
@@ -92,34 +94,72 @@ export const SegmentedControl: React.FC<SegmentedControlProps> = React.forwardRe
         value: controlledValue,
         ...htmlProps
     } = props;
+    const [selectedIndex, setSelectedIndex] = React.useState<number>(
```

Comment:
we should keep this code more like it was before - where we use this local state index only if using the `defaultValue` option to avoid duplicating state, then on every render reconcile the index of the user provided `controlledValue` with this local index as a fallback

Reply:
Hmm it's tricky because we want to add keypress control with this PR, which is much easier to change to the next index vs. the next value. I can take a shot at it though.

Reply:
Done! Reduced the diff a ton, mostly the same code as was before, but arrow keys work as indended.